bottlecap/src/logs/flusher.rs

```diff
@@ -196,3 +182,88 @@ impl Flusher {
         encoder.finish().map_err(|e| Box::new(e) as Box<dyn Error>)
     }
 }
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Clone)]
+pub struct LogsFlusher {
+    pub flushers: Vec<Flusher>,
+}
+
+impl LogsFlusher {
+    pub fn new(
+        api_key: String,
+        aggregator: Arc<Mutex<Aggregator>>,
+        config: Arc<config::Config>,
+    ) -> Self {
+        let mut flushers = Vec::new();
+
+        // Create primary flusher
+        flushers.push(Flusher::new(
+            api_key.clone(),
+            config.logs_config_logs_dd_url.clone(),
+            aggregator.clone(),
+            config.clone(),
+        ));
+
+        // Create flushers for additional endpoints
+        for endpoint in &config.logs_config_additional_endpoints {
+            let endpoint_url = format!("https://{}:{}", endpoint.host, endpoint.port);
+            flushers.push(Flusher::new(
+                endpoint.api_key.clone(),
+                endpoint_url,
+                aggregator.clone(),
+                config.clone(),
+            ));
+        }
+
+        LogsFlusher { flushers }
+    }
+
+    pub async fn flush(
+        &self,
+        retry_request: Option<reqwest::RequestBuilder>,
+    ) -> Vec<reqwest::RequestBuilder> {
+        let mut failed_requests = Vec::new();
+
+        // If retry_request is provided, only process that request
+        if let Some(req) = retry_request {
+            if let Some(req_clone) = req.try_clone() {
+                if let Err(e) = Flusher::send(req_clone).await {
+                    if let Some(failed_req_err) = e.downcast_ref::<FailedRequestError>() {
+                        failed_requests.push(
+                            failed_req_err
+                                .request
+                                .try_clone()
+                                .expect("should be able to clone request"),
+                        );
+                    }
+                }
+            }
+        } else {
+            // Get batches from primary flusher's aggregator
+            let logs_batches = {
+                let mut guard = self.flushers[0].aggregator.lock().expect("lock poisoned");
+                let mut batches = Vec::new();
+                let mut current_batch = guard.get_batch();
+                while !current_batch.is_empty() {
+                    batches.push(current_batch);
+                    current_batch = guard.get_batch();
+                }
+                batches
+            };
+
+            // Send batches to each flusher
+            let futures = self.flushers.iter().map(|flusher| {
+                let batches = logs_batches.clone();
+                let flusher = flusher.clone();
+                async move { flusher.flush(Some(batches)).await }
+            });
+
+            let results = join_all(futures).await;
+            for failed in results {
+                failed_requests.extend(failed);
+            }
+        }
```

Comment:
Would it be possible to create the logs_batches, pass as a reference to all flushers, and then drop when used by all flushers? That way we don't clone? 

Not sure if you have answered this question before for metrics, I imagine maybe we do need a copy to send it?

Just LMK! Trying to reduce the memory footprint here